pkg/resources/management.cattle.io/v3/clusterroletemplatebinding/validator.go

```diff
@@ -130,50 +120,58 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
-		invalidFieldName = "userPrincipalName"
+		return field.Invalid(fieldPath.Child("userPrincipalName"), newCRTB.UserPrincipalName, reason)
 	case oldCRTB.GroupName != newCRTB.GroupName && oldCRTB.GroupName != "":
-		invalidFieldName = "groupName"
+		return field.Invalid(fieldPath.Child("groupName"), newCRTB.GroupName, reason)
 	case oldCRTB.GroupPrincipalName != newCRTB.GroupPrincipalName && oldCRTB.GroupPrincipalName != "":
-		invalidFieldName = "groupPrincipalName"
+		return field.Invalid(fieldPath.Child("groupPrincipalName"), newCRTB.GroupPrincipalName, reason)
 	case (newCRTB.GroupName != "" || oldCRTB.GroupPrincipalName != "") && (newCRTB.UserName != "" || oldCRTB.UserPrincipalName != ""):
-		invalidFieldName = "both user and group"
+		return field.Forbidden(fieldPath,
+			"binding target must target either a user [userName]/[userPrincipalName] OR a group [groupName]/[groupPrincipalName]")
 	default:
 		return nil
 	}
-
-	return fmt.Errorf("cannot update %s for clusterRoleTemplateBinding %s: %w", invalidFieldName, oldCRTB.Name, admission.ErrInvalidRequest)
 }
 
 // validateCreateFields checks if all required fields are present and valid.
-func (a *admitter) validateCreateFields(newCRTB *apisv3.ClusterRoleTemplateBinding) error {
+func (a *admitter) validateCreateFields(newCRTB *apisv3.ClusterRoleTemplateBinding, fieldPath *field.Path) error {
+	const reason = "field is required"
+
 	hasUserTarget := newCRTB.UserName != "" || newCRTB.UserPrincipalName != ""
 	hasGroupTarget := newCRTB.GroupName != "" || newCRTB.GroupPrincipalName != ""
 
 	if (hasUserTarget && hasGroupTarget) || (!hasUserTarget && !hasGroupTarget) {
-		return fmt.Errorf("binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]: %w", admission.ErrInvalidRequest)
+		return field.Forbidden(fieldPath, "binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]")
 	}
 
 	if newCRTB.ClusterName == "" {
-		return fmt.Errorf("missing required field 'clusterName': %w", admission.ErrInvalidRequest)
+		return field.Required(fieldPath.Child("clusterName"), reason)
+	}
+
+	if newCRTB.RoleTemplateName == "" {
+		return field.Required(fieldPath.Child("roleTemplateName"), reason)
 	}
 
 	roleTemplate, err := a.roleTemplateResolver.RoleTemplateCache().Get(newCRTB.RoleTemplateName)
 	if err != nil {
-		return fmt.Errorf("unknown reference roleTemplate '%s': %w", newCRTB.RoleTemplateName, err)
+		if apierrors.IsNotFound(err) {
+			return field.Invalid(fieldPath.Child("roleTemplateName"), newCRTB.RoleTemplateName, "the referenced role template was not found")
+		}
+		return err
 	}
 
 	if roleTemplate.Locked {
-		return fmt.Errorf("referenced role '%s' is locked and cannot be assigned: %w", roleTemplate.DisplayName, admission.ErrInvalidRequest)
+		return field.Forbidden(fieldPath.Child("roleTemplate"), "the referenced role template cannot be locked")
```

Comment:
nit:
I would add some more information so the users know what a locked roletemplate means.